File: app/src/main/java/com/bq/zowi/controllers/AchievementsController.java
package com.bq.zowi.controllers;

import com.bq.zowi.models.Achievement;
import java.util.ArrayList;
import rx.Single;

/* JADX INFO: loaded from: classes.dex */
public interface AchievementsController {
    Single<Achievement> getAchievement(Achievement.Id id);

    Single<ArrayList<Achievement>> getAchievementsList();

    void resetAchievementsList();

    Single<Achievement> unlockAchievement(Achievement.Id id);

    Single<Void> unlockAllAchievements();
}

File: zowi-core/src/main/kotlin/com/bq/zowi/api/AchievementsController.kt
package com.bq.zowi.api

import com.bq.zowi.models.Achievement
import io.reactivex.Single

interface AchievementsController {
    fun getAchievement(id: Achievement.Id): Single<Achievement>
    fun getAchievementsList(): Single<ArrayList<Achievement>>
    fun resetAchievementsList()
    fun unlockAchievement(id: Achievement.Id): Single<Achievement>
    fun unlockAllAchievements(): Single<Void>
}
